sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -134,8 +137,12 @@ public URL getQueueUrl() {
      * @return A response that only contains headers and response status code
      * @throws StorageErrorException If a queue with the same name and different metadata already exists in the queue service.
      */
-    public Mono<VoidResponse> create(Map<String, String> metadata) {
-        return client.queues().createWithRestResponseAsync(queueName, null, metadata, null, Context.NONE)
+    public Mono<VoidResponse> createWithResponse(Map<String, String> metadata) {
+        return withContext(context -> createWithResponse(metadata, context));
+    }
+
+    Mono<VoidResponse> createWithResponse(Map<String, String> metadata, Context context) {
```

Comment:
Not quite understand why we make the API protected.
If there is no need to have context in async one. Can we make it private?

Reply:
This is being called from the sync client methods in which we need explicit context argument passing, so protected.

Reply:
This is 'package-private', not 'protected'. This means it is not public API and effectively private.